### ISSUE TITLE
[Issue]: Strange behaviour of collections

### ISSUE THREAD

**Issue by ghost** [jellyfin/jellyfin#11594]
### Please describe your bug

In 10.9 collections were broken and "fixed" for 10.9.1 (#11572)
However now around 1 in 6 items I add to a collection do not show up in it.

### Reproduction Steps

Add multiple items to a collection
Check if all items are in the collection.
Some are missing!

### Jellyfin Version

Master branch

### if other:

10.9.1

### Environment

```markdown
Jellyfin Server on Win 10 Education
i7 6500U 16 GB RAM
GTX 950M
```

### Jellyfin logs

```shell
None
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by space-seed** [comment]
Ok, i've done some further testing. The issue mainly persist with one item now in my Alien Collection I cannot add the title "Alien: Covenant" everything else works so far...

**Comment by gnattu** [comment]
Please be clear about it. Is it only one movie not able to be added to a collection right?

**Comment by space-seed** [comment]
So far it is only one movie: "Alien: Covenant". I had a similiar bug in a previous version of Jellyfin, dont know which, where I could use a workaround if a movie would not add to a collection by renaming it, adding it, and then naming it back.

But here so far this is the one that persists no matter what I try.

**Comment by space-seed** [comment]
Ok, Jellyfin wont add "Star Trek II: The Wrath of Khan" to my Star Trek Collection either.

**Comment by gnattu** [comment]
So it seems like Jellyfin don’t want to add movies with titles contain `:` into the collection right? Can you verify that?

**Comment by space-seed** [comment]
No, I could add "Resident Evil: Retribution" and "Resident Evil: The Final Chapter" just fine.
Also after  just adding a few letters to wrath of khan I was able to add it and remove added letters after I added it.
Still no luck with "Alien: Covenant" though...

**Comment by space-seed** [comment]
After renaming the source file (and folder) on my data drive ("Alien_ Covenant (2017)" to "Alien Covenant (2017)") and rescanning the library I was able to add the movie successfully to a collection. 

However I dont believe this necessity is expected behaviour. 
Furthermore I have many other movies where in the filename I use the _ which all behave as expected.

**Comment by TheHYPO35** [comment]
I have the same issue with two films in one of my collections. 

This began when I noticed that some of my films had a little movie camera icon in the corner of the thumbnail (or on firestick, a filmstrip icon), and when I go to "edit metadata" on those files, the first field is "content type": inherit (these are films in the same Movie (non-mixed type) library as the other films in the same collections).

When I removed these two films from the collection to see if I could re-add them to solve the problem, the films simply don't show up at all now.

I have tried moving the media files away and then manually "deleting" the films from Jellyfin, then moving the files back and rescanning (this worked for me a year or so ago to completely clear the films from my jellyfin database when I had other films that similarly wouldn't appear in collections) but this does not work for the current problem.

**Comment by space-seed** [comment]
Ar you also on Windows?I think its something to do with failed items in the collection. After changing the filename on the drive ie giving it a ever so slightly but still different path, I could add my movies.

**Comment by TheHYPO35** [comment]
This is a Linux docker install. I have tried renaming the file without success. I haven’t tried renaming the folder which is in a consistent format, so changing it would mess up my organization, but I suppose I can do it temporarily at least to test.

**Comment by danni-storm** [comment]
I have also have issues with seemingly no pattern. Commonly it's movies with a singular title which is part of all of the series item titles (e.g. Star Wars or Star Trek). However, this isn't always the case such as my Batman collection where some titles do not have Batman in the title. 

Running the "_Clean up collections and playlists_" task does not fix the issue, nor does deleting the collection from the web UI and/or filesystem `collection.xml`. No errors are reported on failed add or cleanup and when adding items there is a confirmation box that the item (which was not added into the UI) was 'successfully added'. This makes sense when looking at the `xml` as it does indeed add the item multiple times into the `collection.xml`.

Generally, collections accept anywhere from 1 to 6 items and then stop presenting them in the UI. `xml` file saving is enabled and, currently, Jellfyin has `rw` access to the media folders. Automatic collection generation is disabled at the library level and `TMDb Box Sets` is disabled. 

This is the contents of the Batman `collection.xml` (Batman Begins is the problem file that will not add into the ui):
```
<?xml version="1.0" encoding="utf-8" standalone="yes"?>
<Item>
  <ContentRating>PG-13</ContentRating>
  <Added>05/02/2024 08:16:17</Added>
  <LockData>false</LockData>
  <Overview>Batman is the alias of Bruce Wayne, a wealthy American playboy, philanthropist, and industrialist who resides in Gotham City. Batman's origin story features him swearing vengeance against criminals after witnessing the murder of his parents Thomas and Martha as a child, a vendetta tempered with the ideal of justice. He trains himself physically and intellectually, crafts a bat-inspired persona, and monitors the Gotham streets at night.</Overview>
  <LocalTitle>Batman Collection</LocalTitle>
  <PremiereDate>1989-06-20</PremiereDate>
  <DisplayOrder>PremiereDate</DisplayOrder>
  <ProductionYear>1989</ProductionYear>
  <TmdbId>120794</TmdbId>
  <Genres>
    <Genre>Drama</Genre>
    <Genre>Action</Genre>
    <Genre>Crime</Genre>
    <Genre>Thriller</Genre>
    <Genre>Fantasy</Genre>
    <Genre>Mystery</Genre>
    <Genre>Adventure</Genre>
  </Genres>
  <Studios>
    <Studio>DC Comics</Studio>
    <Studio>Legendary Pictures</Studio>
    <Studio>Syncopy</Studio>
    <Studio>Isobel Griffiths</Studio>
    <Studio>Warner Bros. Pictures</Studio>
    <Studio>DC Entertainment</Studio>
    <Studio>Polygram Pictures</Studio>
    <Studio>6th &amp; Idaho</Studio>
    <Studio>Dylan Clark Productions</Studio>
    <Studio>DC Films</Studio>
    <Studio>The Guber-Peters Company</Studio>
    <Studio>RatPac Entertainment</Studio>
    <Studio>Atlas Entertainment</Studio>
    <Studio>Cruel &amp; Unusual Films</Studio>
  </Studios>
  <CollectionItems>
    <CollectionItem>
      <Path>/media/Movies/The Dark Knight (2008) [imdbid-tt0468569]/The Dark Knight (2008) [imdbid-tt0468569] - 1080p.mkv</Path>
    </CollectionItem>
    <CollectionItem>
      <Path>/media/Movies/The Dark Knight Rises (2012) [imdbid-tt1345836]/The Dark Knight Rises (2012) [imdbid-tt1345836] - 1080p.mkv</Path>
    </CollectionItem>
    <CollectionItem>
      <Path>/media/Movies/Batman Returns (1992) [imdbid-tt0103776]/Batman Returns (1992) [imdbid-tt0103776] - 4K.mkv</Path>
    </CollectionItem>
    <CollectionItem>
      <Path>/media/Movies/The Batman (2022) [imdbid-tt1877830]/The Batman (2022) [imdbid-tt1877830] - 4K.mkv</Path>
    </CollectionItem>
    <CollectionItem>
      <Path>/media/Movies/Batman (1989) [imdbid-tt0096895]/Batman (1989) [imdbid-tt0096895] - 4K.mkv</Path>
    </CollectionItem>
    <CollectionItem>
      <Path>/media/Movies/Batman Begins (2005) [imdbid - tt0372784]/Batman Begins (2005) [imdbid - tt0372784] - 1080p.mkv</Path>
    </CollectionItem>
    <CollectionItem>
      <Path>/media/Movies/Batman Begins (2005) [imdbid - tt0372784]/Batman Begins (2005) [imdbid - tt0372784] - 1080p.mkv</Path>
    </CollectionItem>
    <CollectionItem>
      <Path>/media/Movies/Batman v Superman - Dawn of Justice (2016) [imdbid-tt2975590]/Batman v Superman - Dawn of Justice (2016) [imdbid-tt2975590] - 4K.mkv</Path>
    </CollectionItem>
    <CollectionItem>
      <Path>/media/Movies/Batman Begins (2005) [imdbid - tt0372784]/Batman Begins (2005) [imdbid - tt0372784] - 1080p.mkv</Path>
    </CollectionItem>
    <CollectionItem>
      <Path>/media/Movies/Batman Begins (2005) [imdbid - tt0372784]/Batman Begins (2005) [imdbid - tt0372784] - 1080p.mkv</Path>
    </CollectionItem>
    <CollectionItem>
      <Path>/media/Movies/Batman Begins (2005) [imdbid - tt0372784]/Batman Begins (2005) [imdbid - tt0372784] - 1080p.mkv</Path>
    </CollectionItem>
    <CollectionItem>
      <Path>/media/Movies/Batman Begins (2005) [imdbid - tt0372784]/Batman Begins (2005) [imdbid - tt0372784] - 1080p.mkv</Path>
    </CollectionItem>
    <CollectionItem>
      <Path>/media/Movies/Batman Begins (2005) [imdbid - tt0372784]/Batman Begins (2005) [imdbid - tt0372784] - 1080p.mkv</Path>
    </CollectionItem>
    <CollectionItem>
      <Path>/media/Movies/Batman Begins (2005) [imdbid - tt0372784]/Batman Begins (2005) [imdbid - tt0372784] - 1080p.mkv</Path>
    </CollectionItem>
  </CollectionItems>
```

**Comment by aerialhedgehog** [comment]
Hi Everyone,

I'm encountering this same issue after upgrading my jellyfin server from 10.8.X to 10.9.8 
- Linuxserver.io 10.9.8ubu2204-ls24
- https://github.com/linuxserver/docker-jellyfin/releases/tag/10.9.8ubu2204-ls24
this is a docker deployment on ubuntu22

In particular for my deployment, I am finding that a Collection of Movies that have **multiple Versions** will struggle to appear in the Collection's browse view.  

## TLDR

I am able to hot fix this issue by modifying the collections.xml file of the Collection to use the ```ItemId```, instead of the ```Path```.  The item id is the UUID that you can extract from the web-viewer's URL, in the id tag  ```details?id=XXXXXXXXXX...```.

My guess after only browsing code for a few hours: the Child ID Linking method has changed such that the discovered ItemId is no longer stable based on a specific path to a specific file.  It might be more stable to add a ItemId to the collections.xml file now, instead of the title Path, at the time the title is added to the collection using the UI.


## Detailed Discussion

In my media library, I have multiple versions of a movie, named based on the recommendation in the documentation [here](https://jellyfin.org/docs/general/server/media/movies/#multiple-versions-of-a-movie).  As an example:
```
media/videos/movies/Harry Potter 4 - Goblet Of Fire/
    backdrop.jpg                                                   
    landscape.jpg
    logo.png
    folder.jpg                                                     
    movie.nfo
    'Harry Potter 4 - Goblet Of Fire - Source1 - 1080p - x264.nfo'   
    'Harry Potter 4 - Goblet Of Fire - Source2 - 1080p - x264.mkv'
    'Harry Potter 4 - Goblet Of Fire - Source1 - 1080p - x264.mp4'   
```

quick note, which turns out not to be related to this issue:
- i am using the nfo metadata savers option, which is where the nfo files have come from
- for some reason the metadatasaver produces a redundant nfo file named after the first version (Source 1 above)
- manipulating the nfo files in this folder did Not affect any useful changes on the Collection view


In 10.8, jellyfin correctly rendered all movies of a collection with the appropriate number of versions in a blue circle in the top left of movie the poster.  

In 10.9(.8), there are three different cases of behavior, which as some above have also mentioned, seem to be random on a movie by movie basis:
1. An individual movie correctly appears in the collection, if there are multiple versions then with the blue number of versions
2. An individual movie appears in the collection, but this only represents a specific file.  The poster will have a movie camera or movie film icon in a gray circle in the top right of the poster.  Clicking through this item will bring you to a page specific to that file name, and as a result it incorrectly renders the various metadata of the movie (i.e. title is wrong and instead parsed from the filename)
3. An individual movie does not appear at all in the collection.

There are a few different ways to add the movie to the collection to yield slightly different results:
- Add the movie from a search view > get random results scenarios 1,2,3
- Add the movie from the library view > get random results scenarios 1,2,3
- Open the movie title, make sure the source is set to the first one > most probably get scenario 2 above (individual movie file)

Seeing that collection.xml files were being populated with paths, I went looking through the XML parser for the collections meta data, hoping that a UUID-like tag was available instead.  There is.  For those stumbling on this and are curious:
- [BoxSetXmlParser](https://github.com/jellyfin/jellyfin/blob/master/MediaBrowser.LocalMetadata/Parsers/BoxSetXmlParser.cs#L66)
- [BaseItemXmlParser](https://github.com/jellyfin/jellyfin/blob/master/MediaBrowser.LocalMetadata/Parsers/BaseItemXmlParser.cs#L799)

Changing the collections list to lists of ItemId results in all movies of multiple versions being correctly rendered in the collection view.


## Additional Debugging Data

I'm hoping someone who knows the source can pin the issue based on this bug walk :)

### 1. Poorly Behaving Collection

Add title to a fresh collection from a search view.

Collections.xml, note that the second source is always added
![1b](https://github.com/user-attachments/assets/8bc979b7-4bfe-4ec3-87bb-1a6873a3d24c)

Library is now missing two titles
![1a](https://github.com/user-attachments/assets/48a9f163-6854-4c7c-a1c8-10196449f8bf)


### 2. Forcing a title in through the WebUI

Open the movie title, select the first version in the drop down list, then add the title to the collection
![2a](https://github.com/user-attachments/assets/5fccc89c-c89f-454e-a27b-c6ca333a47d9)

Collections.xml, showing the two missing titles changed to paths for the first version of each title
![2b](https://github.com/user-attachments/assets/e1b7599f-f963-414c-959d-2895ff86cf5f)

Library now shows all titles, but the two forced titles go to the specific movie file (the movie icon in the top right of the poster)
![2c](https://github.com/user-attachments/assets/ce943d58-b9d1-4ebc-a61e-1daa94c8533d)


### 3. Using the ItemId instead

Pull the item id from the webUI's URL as below
![3a1](https://github.com/user-attachments/assets/57c135fa-53a3-4202-82ce-bc513c507503)

Updated collections.xml, now with ItemIds. 
![3b](https://github.com/user-attachments/assets/8a5de6fb-1964-4210-bb8d-492e360f0e91)

Collection is now showing the titles correctly again
![3c](https://github.com/user-attachments/assets/cc25c9a8-3d86-4ff9-8ce7-8a1bab493396)

**Comment by danni-storm** [comment]
Interesting! Thanks for looking in to this in more detail. I did think it strange the `xml` using the absolute path instead of itemid or something similar.

**Comment by danni-storm** [comment]
@aerialhedgehog Have you been able to figure out a fix to submit as a pull request? I am personally not nearly familiar enough with the Jellyfin codebase to make an informed fix but I could try my hand at it in the next few weeks.

**Comment by aerialhedgehog** [comment]
@danni-storm unfortunately no i'm not familiar enough with the code to submit a PR yet.  i've been wondering if i should resubmit my debugging as a standalone issue so the main dev's have a chance at seeing it.  i think the fix could be easy.

going to note an extra issue related to the hotfix i mention above:
- as i mentioned i often have multiple versions of a movie (i.e. original, extended edition, different resolution). 
- there is some subtly about the uuid in the case of a grouping of versions and individual versions:
    - each version gets its own uuid
    - the grouping of the multiple versions has yet another unique id
- the result is that using uuid's in collections.xml can be brittle if you end up needing to delete versions down to leaving only one version.  in this case the grouping uuid disappears, and you need to go pull the uuid for the remaining version and manually update the collections.xml (or just use the regular add-to-collection gui interface since there is no ambiguity any more)

would be nice if the add-to-collection interface could identify that you are adding a grouping of versions, which according to the standard folder layout should be identifiable by putting a folder to the grouping of versions in collections.xml, instead of a path to a specific version.

**Comment by danni-storm** [comment]
I wonder if it would be feasible to link uuids of versions to a 'master' uuid that the add to collections function uses instead of the version uuid. This might already exist?

I agree on the devs seeing this but I think opening a new issue is not the best way to go about doing it. I just posted in Matrix to see if we can input from them as older issues, even with a lot of activity, tend to, understandably, fall under the radar.

**Comment by felix920506** [comment]
If you have an idea on how to fix this then feel free to discuss it in our matrix / discord or open a discussion on this repo.
https://jellyfin.org/contact
